GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -1115,23 +1115,35 @@ private void RevisionGridSelectionChanged(object sender, EventArgs e)
                 CommitInfoTabControl.SelectedIndexChanged -= CommitInfoTabControl_SelectedIndexChanged;
                 if (!revisions.Any() || GitRevision.IsArtificial(revisions[0].Guid))
                 {
-                    //Artificial commits cannot show tree (ls-tree) and has no commit info 
+                    //Artificial commits cannot show tree (ls-tree) and has no commit info
                     CommitInfoTabControl.RemoveIfExists(CommitInfoTabPage);
                     CommitInfoTabControl.RemoveIfExists(TreeTabPage);
                     CommitInfoTabControl.RemoveIfExists(GpgInfoTabPage);
 
                     if (_showRevisionInfoNextToRevisionGrid)
                     {
                         RevisionsSplitContainer.Panel2Collapsed = true;
-
                     }
+
+                    // Mono special - must re-select the tab for the content to render..
+                    CommitInfoTabControl.SelectedTab = null;
+                    CommitInfoTabControl.SelectedTab = DiffTabPage;
                 }
                 else
                 {
                     int i = 0;
                     if (!_showRevisionInfoNextToRevisionGrid)
                     {
+                        // Mono special - must re-select the tab for the content to render..
+                        if (!CommitInfoTabControl.TabPages.Contains(CommitInfoTabPage))
```

Comment:
If you have something else than CommitTab before switching, CommitTab will always be selected.

Do this in Mono only (or do not remove/add tabs at all in Mono).
I believe #4254 will work better both in Win/Mono